restdocs-api-spec-jsonschema/src/test/kotlin/com/epages/restdocs/apispec/jsonschema/JsonSchemaFromFieldDescriptorsGeneratorTest.kt

```diff
@@ -289,6 +292,89 @@ class JsonSchemaFromFieldDescriptorsGeneratorTest {
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(CombinedSchema::class.java)
+        val combinedSchema = ((schema as ArraySchema).allItemSchema) as CombinedSchema
+        then(combinedSchema.criterion).isEqualTo(ONE_CRITERION)
+        then(combinedSchema.subschemas).extracting("class").containsExactlyInAnyOrder(
+            ObjectSchema::class.java,
+            BooleanSchema::class.java,
+            StringSchema::class.java,
+            NumberSchema::class.java
+        )
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_specify_accurate_items_type_in_array_when_descriptor_contains_itemsType_in_additionalParameters() {
+        givenFieldDescriptorWithArrayOfSingleType()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(StringSchema::class.java)
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_specify_generic_items_type_in_array_of_array_when_descriptor_doesnt_contain_itemsType_in_additionalParameters() {
+        givenFieldDescriptorWithTopLevelArrayOfArrayOfAny()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(ArraySchema::class.java)
+        val arrayOfArraySchema = ((schema as ArraySchema).allItemSchema) as ArraySchema
+        then(arrayOfArraySchema.allItemSchema).isInstanceOf(CombinedSchema::class.java)
+        val combinedSchema = arrayOfArraySchema.allItemSchema as CombinedSchema
+        then(combinedSchema.criterion).isEqualTo(ONE_CRITERION)
+        then(combinedSchema.subschemas).extracting("class").containsExactlyInAnyOrder(
+            ObjectSchema::class.java,
+            BooleanSchema::class.java,
+            StringSchema::class.java,
+            NumberSchema::class.java
+        )
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_specify_accurate_items_type_in_array_of_array_when_descriptor_contains_itemsType_in_additionalParameters() {
+        givenFieldDescriptorWithTopLevelArrayOfArrayOfSingleType()
+
+        whenSchemaGenerated()
+
+        then(schema).isInstanceOf(ArraySchema::class.java)
+        then((schema as ArraySchema).allItemSchema).isInstanceOf(ArraySchema::class.java)
+        val arrayOfArraySchema = (schema as ArraySchema).allItemSchema as ArraySchema
+        then(arrayOfArraySchema.allItemSchema).isInstanceOf(StringSchema::class.java)
+        thenSchemaIsValid()
+    }
+
+    @Test
+    fun should_not_erased_array_content() {
+        fieldDescriptors = listOf(
+            FieldDescriptor("thisIsAnArray", "I'm an array", "ARRAY"),
+            FieldDescriptor("thisIsAnArray[].numberItem", "I'm a number", "NUMBER"),
+            FieldDescriptor("thisIsAnArray[].objectItem", "I'm an object", "OBJECT")
+        )
+        whenSchemaGenerated()
```

Comment:
Please separate the "arrange" and "act" steps by a blank line (add a newline above this line) to make it easier to distinguish the parts.

Reply:
ok done